From 04310c4e59adfa44d6e187367c3b42b8e08c2a51 Mon Sep 17 00:00:00 2001
From: arathinx <ayyappanx.rathinam@intel.com>
Date: Fri, 4 Oct 2019 18:13:36 +0800
Subject: [PATCH] PONRTSYS-2776 :Initial PLL3 config breaks PON IP register
 access on cold boots

In some board, Uboot configured LJPLL3 for Aquantia PHY. CGU driver has to reset the LD bit default value 0 before apply the clock changes
---
 drivers/clk/intel/clk-cgu-pll.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/clk/intel/clk-cgu-pll.c b/drivers/clk/intel/clk-cgu-pll.c
index eb4cac8ddc3a..06e98961da43 100644
--- a/drivers/clk/intel/clk-cgu-pll.c
+++ b/drivers/clk/intel/clk-cgu-pll.c
@@ -241,6 +241,12 @@ static int
 prx300_pll_set_params(struct intel_clk_pll *pll, unsigned int mult,
 		      unsigned int div, unsigned int frac)
 {
+	/* set the default value for LD bit */
+	if ((pll->type == TYPE_LJPLL) &&
+	    (intel_get_clk_val(pll->map, pll->reg, 31, 1))) {
+		intel_set_clk_val(pll->map, pll->reg, 31, 1, 0);
+	}
+
 	intel_set_clk_val(pll->map, pll->reg + 0x8, 28, 1, 1);
 	intel_set_clk_val(pll->map, pll->reg + 0x8, 0, 12, mult);
 	intel_set_clk_val(pll->map, pll->reg + 0x8, 18, 6, div);
